File: Ruby/sort/lambda_function.rb
require 'json'

def sort(arr)
    i = 0
    while i < arr.length - 1
    current_elem = arr[i]
    next_elem = arr[i+1]

    if current_elem > next_elem
        arr[i] = next_elem
        arr[i+1] = current_elem
        i = 0
    else
        i += 1
    end

    end
end

def lambda_handler(event:, context:)
    file = File.read(event['filepath'])
    arr = JSON.parse(file)['numbers']
    # TODO implement
    sort(arr)
    { statusCode: 200, body: "Sorted." }
end

File: Python/sort/lambda_function.py
import json

def sort(arr):
    i = 0
    while i < len(arr) - 1:
        current_elem = arr[i]
        next_elem = arr[i + 1]
        if current_elem > next_elem:
            arr[i] = next_elem
            arr[i + 1] = current_elem
            i = 0
            continue
        i += 1
        

def lambda_handler(event, context):
    # TODO implement
    size = event['size']
    data = open(f'pi_input_{size}.json')
    arr = json.load(data)['numbers']
    sort(arr)
    return {
        'statusCode': 200,
        'body': "Sorted."
    }
